server/server.go

```diff
@@ -229,8 +245,42 @@ func (s *server) Boot() {
 			}(e)
 		}
 
-		// Register prometheus metrics endpoint.
-		s.router.Path("/metrics").Handler(promhttp.Handler())
+		// If the user provided a specific url for the metrics endpoint:
+		if s.listenMetricsUrl != nil {
+			// Register prometheus metrics endpoint to a different server as the rest
+			// of the endpoints.
+			go func() {
+				s.logger.Log("level", "debug", "message", fmt.Sprintf("running metrics server at %s", s.listenMetricsUrl.String()))
+
+				metricsRouter := mux.NewRouter()
+				metricsRouter.Path("/metrics").Handler(promhttp.Handler())
+
+				s.metricsHTTPServer = &http.Server{
+					Addr:    s.listenMetricsUrl.Host,
+					Handler: metricsRouter,
+				}
+
+				if s.listenMetricsUrl.Scheme == "https" {
+					tlsConfig, err := tls.LoadTLSConfig(s.tlsCertFiles)
```

Comment:
Depending on used configuration, I'd say that most often certs provided for main server don't work for metrics server (because it's bound to different address and most probably has different DNS name if there is one at all). In order this to work, there would need to be SAN entry for metrics address.

Reply:
good point... maybe we leave this part out? So when there is a listen address for metrics, it _cant_ be over https?

Reply:
Either add separate configuration options for metrics TLS keys or leave it out. I'm fine with both approaches.